table/block.cc

```diff
@@ -327,7 +327,9 @@ bool DataBlockIter::SeekForGet(const Slice& target) {
   // Here we are conservative and only support a limited set of cases
   ValueType value_type = ExtractValueType(key_.GetKey());
   if (value_type != ValueType::kTypeValue &&
-      value_type != ValueType::kTypeDeletion) {
+      value_type != ValueType::kTypeDeletion &&
+      value_type != ValueType::kTypeSingleDeletion &&
+      value_type != ValueType::kTypeBlobIndex) {
```

Comment:
Would it be better to check for inclusion, instead of exclusion? Otherwise I think we might have a whole bunch to exclude here. That also feels much safer to me, future-proofing the code when we add new types. 
Something like:
```
(value_type == ValueType::kTypeValue || value_type == something else)
```